server/src/controllers/ResourcesController.js

```diff
@@ -1,22 +1,38 @@
 const { Resource, Topic } = require("../models/Resources");
+const redis_client = require("../config/redis");
+
+/* 
+  Topics and resources behave a bit differently when it comes to endpoints. The delete cache
+  middleware does a path lookup, but we need to update the "topics" values, even if 
+  there is a change in resources. So, a function that does the same tasks, but exclusively for topics.
+*/
+function deleteCache(req, res) {
+  redis_client.get("topics", (error, reply) => {
+    if (error) {
+      return res.status(500).json({ error });
```

Comment:
If the key doesn't exist in the cache, then the function should simply return rather than sending a `500` response.

Reply:
The delete cache function should ideally not have req and res as parameters.